### ISSUE TITLE
[Device Support Request] ThirdReality Multifunction Light 3RSNL02043Z

### ISSUE THREAD

**Issue by llhappier** [zigpy/zha-device-handlers#2461]
### Problem description

Hi, 
I have a multifunction night light. It use private motion cluster and I want it work correctly.
By the way, how can i remove the start-up color temprature? Thanks.

### Solution description

I think it should it should change private cluster(0xfc00) to ias zone(0x0500) .

### Device signature
<details><summary>Device signature</summary>

```json
{
  "node_descriptor": "NodeDescriptor(logical_type=<LogicalType.Router: 1>, complex_descriptor_available=0, user_descriptor_available=0, reserved=0, aps_flags=0, frequency_band=<FrequencyBand.Freq2400MHz: 8>, mac_capability_flags=<MACCapabilityFlags.AllocateAddress|RxOnWhenIdle|MainsPowered|FullFunctionDevice: 142>, manufacturer_code=4877, maximum_buffer_size=127, maximum_incoming_transfer_size=242, server_mask=11264, maximum_outgoing_transfer_size=242, descriptor_capability_field=<DescriptorCapability.NONE: 0>, *allocate_address=True, *is_alternate_pan_coordinator=False, *is_coordinator=False, *is_end_device=False, *is_full_function_device=True, *is_mains_powered=True, *is_receiver_on_when_idle=True, *is_router=True, *is_security_capable=False)",
  "endpoints": {
    "1": {
      "profile_id": "0x0104",
      "device_type": "0x0102",
      "input_clusters": [
        "0x0000",
        "0x0003",
        "0x0004",
        "0x0005",
        "0x0006",
        "0x0008",
        "0x0012",
        "0x0300",
        "0x0400",
        "0x1000",
        "0xfc00"
      ],
      "output_clusters": [
        "0x0019"
      ]
    }
  },
  "manufacturer": "Third Reality, Inc",
  "model": "3RSNL02043Z"
}
```

</details>
 

### Diagnostic information
<details><summary>Diagnostic information</summary>

```json
{
  "home_assistant": {
    "installation_type": "Home Assistant OS",
    "version": "2023.5.4",
    "dev": false,
    "hassio": true,
    "virtualenv": false,
    "python_version": "3.10.11",
    "docker": true,
    "arch": "aarch64",
    "timezone": "Asia/Hong_Kong",
    "os_name": "Linux",
    "os_version": "6.1.21-v8",
    "supervisor": "2023.07.1",
    "host_os": "Home Assistant OS 10.2",
    "docker_version": "23.0.6",
    "chassis": "embedded",
    "run_as_root": true
  },
  "custom_components": {},
  "integration_manifest": {
    "domain": "zha",
    "name": "Zigbee Home Automation",
    "after_dependencies": [
      "onboarding",
      "usb"
    ],
    "codeowners": [
      "@dmulcahey",
      "@adminiuga",
      "@puddly"
    ],
    "config_flow": true,
    "dependencies": [
      "file_upload"
    ],
    "documentation": "https://www.home-assistant.io/integrations/zha",
    "iot_class": "local_polling",
    "loggers": [
      "aiosqlite",
      "bellows",
      "crccheck",
      "pure_pcapy3",
      "zhaquirks",
      "zigpy",
      "zigpy_deconz",
      "zigpy_xbee",
      "zigpy_zigate",
      "zigpy_znp"
    ],
    "requirements": [
      "bellows==0.35.5",
      "pyserial==3.5",
      "pyserial-asyncio==0.6",
      "zha-quirks==0.0.99",
      "zigpy-deconz==0.21.0",
      "zigpy==0.55.0",
      "zigpy-xbee==0.18.0",
      "zigpy-zigate==0.11.0",
      "zigpy-znp==0.11.1"
    ],
    "usb": [
      {
        "vid": "10C4",
        "pid": "EA60",
        "description": "*2652*",
        "known_devices": [
          "slae.sh cc2652rb stick"
        ]
      },
      {
        "vid": "1A86",
        "pid": "55D4",
        "description": "*sonoff*plus*",
        "known_devices": [
          "sonoff zigbee dongle plus v2"
        ]
      },
      {
        "vid": "10C4",
        "pid": "EA60",
        "description": "*sonoff*plus*",
        "known_devices": [
          "sonoff zigbee dongle plus"
        ]
      },
      {
        "vid": "10C4",
        "pid": "EA60",
        "description": "*tubeszb*",
        "known_devices": [
          "TubesZB Coordinator"
        ]
      },
      {
        "vid": "1A86",
        "pid": "7523",
        "description": "*tubeszb*",
        "known_devices": [
          "TubesZB Coordinator"
        ]
      },
      {
        "vid": "1A86",
        "pid": "7523",
        "description": "*zigstar*",
        "known_devices": [
          "ZigStar Coordinators"
        ]
      },
      {
        "vid": "1CF1",
        "pid": "0030",
        "description": "*conbee*",
        "known_devices": [
          "Conbee II"
        ]
      },
      {
        "vid": "10C4",
        "pid": "8A2A",
        "description": "*zigbee*",
        "known_devices": [
          "Nortek HUSBZB-1"
        ]
      },
      {
        "vid": "0403",
        "pid": "6015",
        "description": "*zigate*",
        "known_devices": [
          "ZiGate+"
        ]
      },
      {
        "vid": "10C4",
        "pid": "EA60",
        "description": "*zigate*",
        "known_devices": [
          "ZiGate"
        ]
      },
      {
        "vid": "10C4",
        "pid": "8B34",
        "description": "*bv 2010/10*",
        "known_devices": [
          "Bitron Video AV2010/10"
        ]
      }
    ],
    "zeroconf": [
      {
        "type": "_esphomelib._tcp.local.",
        "name": "tube*"
      },
      {
        "type": "_zigate-zigbee-gateway._tcp.local.",
        "name": "*zigate*"
      },
      {
        "type": "_zigstar_gw._tcp.local.",
        "name": "*zigstar*"
      },
      {
        "type": "_slzb-06._tcp.local.",
        "name": "slzb-06*"
      }
    ],
    "is_built_in": true
  },
  "data": {
    "ieee": "**REDACTED**",
    "nwk": 49749,
    "manufacturer": "Third Reality, Inc",
    "model": "3RSNL02043Z",
    "name": "Third Reality, Inc 3RSNL02043Z",
    "quirk_applied": false,
    "quirk_class": "zigpy.device.Device",
    "manufacturer_code": 4877,
    "power_source": "Mains",
    "lqi": 255,
    "rssi": null,
    "last_seen": "2023-07-06T21:43:16",
    "available": true,
    "device_type": "Router",
    "signature": {
      "node_descriptor": "NodeDescriptor(logical_type=<LogicalType.Router: 1>, complex_descriptor_available=0, user_descriptor_available=0, reserved=0, aps_flags=0, frequency_band=<FrequencyBand.Freq2400MHz: 8>, mac_capability_flags=<MACCapabilityFlags.AllocateAddress|RxOnWhenIdle|MainsPowered|FullFunctionDevice: 142>, manufacturer_code=4877, maximum_buffer_size=127, maximum_incoming_transfer_size=242, server_mask=11264, maximum_outgoing_transfer_size=242, descriptor_capability_field=<DescriptorCapability.NONE: 0>, *allocate_address=True, *is_alternate_pan_coordinator=False, *is_coordinator=False, *is_end_device=False, *is_full_function_device=True, *is_mains_powered=True, *is_receiver_on_when_idle=True, *is_router=True, *is_security_capable=False)",
      "endpoints": {
        "1": {
          "profile_id": "0x0104",
          "device_type": "0x0102",
          "input_clusters": [
            "0x0000",
            "0x0003",
            "0x0004",
            "0x0005",
            "0x0006",
            "0x0008",
            "0x0012",
            "0x0300",
            "0x0400",
            "0x1000",
            "0xfc00"
          ],
          "output_clusters": [
            "0x0019"
          ]
        }
      },
      "manufacturer": "Third Reality, Inc",
      "model": "3RSNL02043Z"
    },
    "active_coordinator": false,
    "entities": [
      {
        "entity_id": "button.third_reality_inc_3rsnl02043z_identify",
        "name": "Third Reality, Inc 3RSNL02043Z"
      },
      {
        "entity_id": "light.third_reality_inc_3rsnl02043z_light",
        "name": "Third Reality, Inc 3RSNL02043Z"
      },
      {
        "entity_id": "number.third_reality_inc_3rsnl02043z_on_off_transition_time",
        "name": "Third Reality, Inc 3RSNL02043Z"
      },
      {
        "entity_id": "number.third_reality_inc_3rsnl02043z_on_level",
        "name": "Third Reality, Inc 3RSNL02043Z"
      },
      {
        "entity_id": "number.third_reality_inc_3rsnl02043z_start_up_current_level",
        "name": "Third Reality, Inc 3RSNL02043Z"
      },
      {
        "entity_id": "number.third_reality_inc_3rsnl02043z_start_up_color_temperature",
        "name": "Third Reality, Inc 3RSNL02043Z"
      },
      {
        "entity_id": "select.third_reality_inc_3rsnl02043z_start_up_behavior",
        "name": "Third Reality, Inc 3RSNL02043Z"
      },
      {
        "entity_id": "sensor.third_reality_inc_3rsnl02043z_illuminance",
        "name": "Third Reality, Inc 3RSNL02043Z"
      }
    ],
    "neighbors": [],
    "routes": [],
    "endpoint_names": [
      {
        "name": "COLOR_DIMMABLE_LIGHT"
      }
    ],
    "user_given_name": null,
    "device_reg_id": "bb132dfeb1abdac7a25c0759eb467580",
    "area_id": null,
    "cluster_details": {
      "1": {
        "device_type": {
          "name": "COLOR_DIMMABLE_LIGHT",
          "id": 258
        },
        "profile_id": 260,
        "in_clusters": {
          "0x0000": {
            "endpoint_attribute": "basic",
            "attributes": {
              "0x0004": {
                "attribute_name": "manufacturer",
                "value": "Third Reality, Inc"
              },
              "0x0005": {
                "attribute_name": "model",
                "value": "3RSNL02043Z"
              }
            },
            "unsupported_attributes": {}
          },
          "0x0003": {
            "endpoint_attribute": "identify",
            "attributes": {},
            "unsupported_attributes": {}
          },
          "0x0004": {
            "endpoint_attribute": "groups",
            "attributes": {},
            "unsupported_attributes": {}
          },
          "0x0005": {
            "endpoint_attribute": "scenes",
            "attributes": {},
            "unsupported_attributes": {}
          },
          "0x0006": {
            "endpoint_attribute": "on_off",
            "attributes": {
              "0x0000": {
                "attribute_name": "on_off",
                "value": 0
              },
              "0x4003": {
                "attribute_name": "start_up_on_off",
                "value": 255
              }
            },
            "unsupported_attributes": {}
          },
          "0x0008": {
            "endpoint_attribute": "level",
            "attributes": {
              "0x0000": {
                "attribute_name": "current_level",
                "value": 254
              },
              "0x0010": {
                "attribute_name": "on_off_transition_time",
                "value": 0
              },
              "0x0011": {
                "attribute_name": "on_level",
                "value": 255
              },
              "0x4000": {
                "attribute_name": "start_up_current_level",
                "value": 255
              }
            },
            "unsupported_attributes": {
              "0x0014": {
                "attribute_name": "default_move_rate"
              },
              "0x0013": {
                "attribute_name": "off_transition_time"
              },
              "0x0012": {
                "attribute_name": "on_transition_time"
              }
            }
          },
          "0x0012": {
            "endpoint_attribute": "multistate_input",
            "attributes": {},
            "unsupported_attributes": {}
          },
          "0x0300": {
            "endpoint_attribute": "light_color",
            "attributes": {
              "0x0000": {
                "attribute_name": "current_hue",
                "value": 169
              },
              "0x0001": {
                "attribute_name": "current_saturation",
                "value": 254
              },
              "0x0003": {
                "attribute_name": "current_x",
                "value": 9843
              },
              "0x0004": {
                "attribute_name": "current_y",
                "value": 3981
              },
              "0x0007": {
                "attribute_name": "color_temperature",
                "value": 65535
              },
              "0x0008": {
                "attribute_name": "color_mode",
                "value": 0
              },
              "0x000f": {
                "attribute_name": "options",
                "value": 1
              },
              "0x4000": {
                "attribute_name": "enhanced_current_hue",
                "value": 43264
              },
              "0x4002": {
                "attribute_name": "color_loop_active",
                "value": 0
              },
              "0x400a": {
                "attribute_name": "color_capabilities",
                "value": 31
              },
              "0x400b": {
                "attribute_name": "color_temp_physical_min",
                "value": 153
              },
              "0x400c": {
                "attribute_name": "color_temp_physical_max",
                "value": 555
              },
              "0x4010": {
                "attribute_name": "start_up_color_temperature",
                "value": 65535
              }
            },
            "unsupported_attributes": {}
          },
          "0x0400": {
            "endpoint_attribute": "illuminance",
            "attributes": {
              "0x0000": {
                "attribute_name": "measured_value",
                "value": 17694
              }
            },
            "unsupported_attributes": {}
          },
          "0x1000": {
            "endpoint_attribute": "lightlink",
            "attributes": {},
            "unsupported_attributes": {}
          },
          "0xfc00": {
            "endpoint_attribute": "manufacturer_specific",
            "attributes": {},
            "unsupported_attributes": {}
          }
        },
        "out_clusters": {
          "0x0019": {
            "endpoint_attribute": "ota",
            "attributes": {},
            "unsupported_attributes": {}
          }
        }
      }
    }
  }
}
```

</details>

### Custom quirk
<details><summary>Custom quirk</summary>

```python
"""Third Reality night light zigbee devices."""
import logging

from zigpy.profiles import zha
from zigpy.quirks import CustomCluster, CustomDevice
from zigpy.zcl.clusters.security import IasZone
from zigpy.zcl.clusters.closures import WindowCovering
from zigpy.zcl.clusters.general import (
    Basic,
    Groups,
    Identify,
    LevelControl,
    MultistateInput,
    OnOff,
    Ota,
    Scenes,
    Time,
)
from zigpy.zcl.clusters.lighting import Color
from zigpy.zcl.clusters.measurement import (
    IlluminanceMeasurement,
)

from zhaquirks import CustomCluster
from zhaquirks.const import (
    DEVICE_TYPE,
    ENDPOINTS,
    INPUT_CLUSTERS,
    MODELS_INFO,
    OUTPUT_CLUSTERS,
    PROFILE_ID,
)

from zhaquirks import (
    Bus,
)

from zhaquirks.const import (
    ATTRIBUTE_ID,
    ATTRIBUTE_NAME,
    COMMAND_ATTRIBUTE_UPDATED,
    COMMAND_TRIPLE,
    UNKNOWN,
    VALUE,
    ZHA_SEND_EVENT,
    ZONE_STATUS,
)

MOTION_REPORTED = "motion_reported"

from zhaquirks.thirdreality import THIRD_REALITY

_LOGGER = logging.getLogger(__name__)

class ThirdRealityMotionCluster(CustomCluster):
    """ThirdRealityMotionCluster."""

    cluster_id = 0xFC00

    def __init__(self, *args, **kwargs):
        """Init."""
        self._current_state = {}
        super().__init__(*args, **kwargs)

    def _update_attribute(self, attrid, value):
        super()._update_attribute(attrid, value)
        #if value is not None and value >= 0:
        self.endpoint.device.motion_bus.listener_event(MOTION_REPORTED, value)

class MotionCluster(CustomCluster):
    """ThirdReality private motion Sensor."""
    cluster_id = 0x0500
    Motion_ID = 0x0002
    def __init__(self, *args, **kwargs):
        """Init."""
        super().__init__(*args, **kwargs)
        self.endpoint.device.motion_bus.add_listener(self)

    def motion_reported(self, value):
        """Power reported."""
        #self._update_attribute(self.Motion_ID, value)
        self.endpoint.ias_zone.update_attribute(self.Motion_ID, value)
    

class Nightlight(CustomDevice):
    """Custom device for 3RSNL02043Z."""
     def __init__(self, *args, **kwargs):
        """Init."""
        self.motion_bus = Bus()
        super().__init__(*args, **kwargs)

    signature = {
        MODELS_INFO: [(THIRD_REALITY, "3RSNL02043Z")],
        ENDPOINTS: {
            1: {
                PROFILE_ID: zha.PROFILE_ID,
                DEVICE_TYPE: zha.DeviceType.COLOR_DIMMABLE_LIGHT,
                INPUT_CLUSTERS: [
                    BasicCluster.cluster_id,
                    Identify.cluster_id,
                    Groups.cluster_id,
                    Scenes.cluster_id,
                    OnOff.cluster_id,
                    LevelControl.cluster_id,
                    MultistateInput.cluster_id,
                    Color.cluster_id,
                    IlluminanceMeasurement.cluster_id,
                    ThirdRealityMotionCluster.cluster_id,
                ],
                OUTPUT_CLUSTERS: [
                    Ota.cluster_id,
                ],
            }
        },
    }
    replacement = {
        SKIP_CONFIGURATION: True,
        ENDPOINTS: {
            1: {
                DEVICE_TYPE: zha.DeviceType.COLOR_DIMMABLE_LIGHT,
                INPUT_CLUSTERS: [
                    BasicCluster.cluster_id,
                    Identify.cluster_id,
                    Groups.cluster_id,
                    Scenes.cluster_id,
                    OnOff.cluster_id,
                    LevelControl.cluster_id,
                    MultistateInput.cluster_id,
                    Color.cluster_id,
                    IlluminanceMeasurement.cluster_id,
                    ThirdRealityMotionCluster,
                    MotionCluster,
                ],
                OUTPUT_CLUSTERS: [
                    Ota.cluster_id,
                ],
            }
        }
    }
```

</details>

**Comment by llhappier** [comment]
In Custom quirk, I also try 

1:
```python
class MotionSensorCluster(IasZone, ThirdRealityLocalCluster):
    """ThirdReality Motion_Sensor Sensor."""

    _CONSTANT_ATTRIBUTES = {ZONE_TYPE: IasZone.ZoneType.Motion_Sensor}
```

2:
```python
class PirMotionManufCluster(ThirdRealityMCUCluster):
    dp_to_attribute: Dict[int, DPToAttributeMapping] = {
        1: DPToAttributeMapping(
            MotionSensorCluster.ep_attribute,
            "zone_status"，
            dp_type=ThirdRealityDPType.BITMAP,
            converter=lambda x: IasZone.ZoneStatus.Alarm_1 if x else 0,
        ),
    }
    data_point_handlers = {
        1: "_dp_2_attr_update",
    }
```

3:
```python
replacement = {
        ENDPOINTS: {
            1: {
                DEVICE_TYPE: zha.DeviceType.COLOR_DIMMABLE_LIGHT,
                INPUT_CLUSTERS: [
                    BasicCluster.cluster_id,
                    Identify.cluster_id,
                    Groups.cluster_id,
                    Scenes.cluster_id,
                    OnOff.cluster_id,
                    LevelControl.cluster_id,
                    MultistateInput.cluster_id,
                    Color.cluster_id,
                    IlluminanceMeasurement.cluster_id,
                    PirMotionManufCluster,
                    MotionSensorCluster,
                ],
                OUTPUT_CLUSTERS: [
                    Ota.cluster_id,
                ],
            }
        }
    }
```

But I dont know how to replace the private cluster(0xfc00) to ias_zone(0x0500). Do somebody have any solutions or suggestions？Thanks.

**Comment by darkxst** [comment]
Does this device work at all for you? I tried without any custom quirk, but it seems to go into a boot loop, going unavailable every 60secs or so.

Edit: I got this working. Occupancy appears to work out of the box (without any custom quirks) on the ssIasZone cluster. Why is this private cluster required and what data does it have? 

I also tested in Z2M and I don't see any occupancy messages coming from this private cluster...
